### PR TITLE
feat(analytics): add `richest-addresses` and `token-distribution` endpoints

### PR DESCRIPTION
**Pull request by Alexandcoats**

### CODE REVIEW

File: bin/inx-chronicle/src/api/stardust/analytics/routes.rs [hunk 1]
```diff
@@ -29,7 +30,8 @@ pub fn routes() -> Router {
             Router::new()
                 .route("/storage-deposit", get(storage_deposit_analytics))
                 .route("/native-tokens", get(unspent_output_analytics::<FoundryOutput>))
-                .route("/nfts", get(unspent_output_analytics::<NftOutput>)),
+                .route("/nfts", get(unspent_output_analytics::<NftOutput>))
+                .route("/richlist", get(richlist_analytics)),
```
Comment (grtlr) [inline bin/inx-chronicle/src/api/stardust/analytics/routes.rs:34]: Nitpick: Probably something like `richest-addresses` is more appropriate? If we change it here, we should also change the other method calls.

File: docs/api-analytics.yml [hunk 2]
```diff
@@ -76,6 +76,26 @@ paths:
           $ref: "#/components/responses/NoResults"
         "500":
           $ref: "#/components/responses/InternalError"
+  /api/analytics/v2/ledger/richlist:
+    get:
+      tags:
+        - ledger
+      summary: Returns a richlist.
```
Comment (grtlr) [inline docs/api-analytics.yml:83]: Reminder.

File: bin/inx-chronicle/src/api/stardust/analytics/responses.rs [hunk 3]
```diff
@@ -50,3 +54,46 @@ pub struct StorageDepositAnalyticsResponse {
...
 
 impl_success_response!(StorageDepositAnalyticsResponse);
+
+/// Response of `GET /api/analytics/richlist[?top=<usize>]`.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RichlistAnalyticsResponse {
+    pub distribution: Vec<DistributionStatDto>,
+    pub top: Vec<AddressStatDto>,
+}
```
Comment (grtlr) [inline bin/inx-chronicle/src/api/stardust/analytics/responses.rs:64]: I think we should split these two queries, following the logic of the other endpoints. One should be `richest-adresses` and the other one should be `token-distribution`.

Additionally, I think we should return the `ledger_index` here too.

File: src/db/collections/outputs/mod.rs [hunk 4]
```diff
@@ -675,3 +675,97 @@ impl MongoDb {
...
+}
+
+impl MongoDb {
+    /// Create richlist statistics.
+    pub async fn get_richlist_analytics(
+        &self,
+        ledger_index: Option<MilestoneIndex>,
+        top: usize,
+    ) -> Result<Richlist, Error> {
+        let ledger_index = match ledger_index {
```
Comment (grtlr) [inline src/db/collections/outputs/mod.rs:710]: `.or_else`?

File: bin/inx-chronicle/src/api/stardust/analytics/routes.rs (hunk 1 above)
Comment (Alexandcoats) [reply bin/inx-chronicle/src/api/stardust/analytics/routes.rs:34]: I dunno, it's more than that. I think richlist is appropriate.

File: docs/api-analytics.yml (hunk 2 above)
Comment (Alexandcoats) [reply docs/api-analytics.yml:83]: Reminder for what?

File: bin/inx-chronicle/src/api/stardust/analytics/responses.rs (hunk 3 above)
Comment (Alexandcoats) [reply bin/inx-chronicle/src/api/stardust/analytics/responses.rs:64]: Fair enough.

File: src/db/collections/outputs/mod.rs (hunk 4 above)
Comment (Alexandcoats) [reply src/db/collections/outputs/mod.rs:710]: Not in _my_ async await

File: docs/api-analytics.yml (hunk 2 above)
Comment (grtlr) [reply docs/api-analytics.yml]: Renaming `richlist` to `richest-addresses` :trollface:

File: src/db/collections/outputs/mod.rs (hunk 4 above)
Comment (grtlr) [reply src/db/collections/outputs/mod.rs:716]: Dang, yeah.

**Review by grtlr — APPROVED** [review]

**Pull request merged**